src/interp/interp-inl.h

```diff
@@ -631,19 +631,19 @@ inline Memory::Ptr Memory::New(interp::Store& store, MemoryType type) {
   return store.Alloc<Memory>(store, type);
 }
 
-inline bool Memory::IsValidAccess(u32 offset, u32 addend, size_t size) const {
-  return u64{offset} + addend + size <= data_.size();
+inline bool Memory::IsValidAccess(u64 offset, u64 addend, u64 size) const {
+  return offset + addend + size <= data_.size();
```

Comment:
Hm, this makes me think we need to be more careful w.r.t. memory accesses that wrap across the 64-bit boundary. It probably needs to check both ends.

Reply:
That in part depends how we define wrapping in the standard, right?
How do we want to do this for now? Check that all 3 components are <=62 bits before the final test? (`size` can be arbitrary as well for one of the bulk memory ops I think).

Reply:
It actually gets interesting: what if `offset + addend` doesn't wrap, but `offset + addend + size` does? :) All pretty hypothetical, but this code does run on every load, so can't be too complicated.